### DISCUSSION TITLE
I have a problem with a Select Menu

### DISCUSSION THREAD

**Discussion by exoristos54** [Rapptz/discord.py#9491] [Q&A] [answered]
I'm new to programming Discord bots and I'm trying to create a Select Menu. I don't understand why it fails despite having all the Privileged Gateway Intents. When interacting with the bot, I get an "Interaction Failed" message. What could be the cause of the error?
here is the code example

    @bot.command()
    async def hello(ctx):
        opts = Select(placeholder='How is the weather today?',
            options=[
                discord.SelectOption(
                    label='sunny',
                    emoji='☀️',
                    description='Sunny day'),
                discord.SelectOption(
                    label='rainy',
                    emoji='🌧️',
                    description='Rainy day'),
            ])
        view = View()
        view.add_item(opts)
        await ctx.send(view=view)

**Comment by LostLuma** [comment, marked as answer]
Hello, to make a Select work properly you'll need to handle the Interaction that is sent to the application when the user has selected something. You can do this by overwriting the [callback](https://discordpy.readthedocs.io/en/stable/interactions/api.html#discord.ui.Select.callback) method in a subclass, there is a [full example here](https://github.com/Rapptz/discord.py/blob/master/examples/views/dropdown.py) alongside other interaction / UI related examples in that same folder.

**Reply by exoristos54** [reply]
thanks

**Reply by No767** [reply]
@s6ql that is simply impossible to remove. It is an discord limitation

**Reply by MCausc78** [reply]
Its possible. However all users will have their options on select reseted too.
```py
class MyView(discord.ui.View):
  @discord.ui.select(options=[
    discord.SelectOption(label='foo', value='1'),
    discord.SelectOption(label='bar', value='2')
  ])
  async def f(self, interaction, _):
    await interaction.response.edit_message(view=self)

await _ctx.reply('Test', view=MyView())
```
Passing `view=self` does that.
